### PR TITLE
linux: enable silead touchscreen driver modules for generic

### PR DESCRIPTION
**Pull request by stpf99**
Hi this is draft for including silead driver and firmware for touchscreen tablets , for review and possible adoption, fetching from fw repo not tested , should be in Libreelec main repo tree cloned, and corresponding changes in package.mk should be adjusted...

### CODE REVIEW

**Comment by chewitt** [comment]
@stpf99 no problem to add support for the driver, but some changes needed:

* Please use =m not =y in the kernel defconfig
* Send the specific firmware for your device to https://github.com/LibreELEC/misc-firmware
* Update `packages/firmware/misc-firmware/package.mk` once the firmware has been merged
* Drop the separate gsl-firmware firmware package (no longer required)
* Rebase the PR so all changes are in a single commit
* Use `package: description of change` for commit message, e.g. `linux: add silead touchscreen driver to generic`

If there is future demand for lots more firmwares we can switch to using a gsl-firmware package, but until that happens we should avoid bloating the image by adding lots of firmwares that won't be used. NB: If we did use a package we would use the original upstream repo and not a fork that has no differences (release tags aren't required for downloading, we use the latest githash for `PKG_VERSION`.

**Comment by stpf99** [comment]
Hi.  fw file for only my device is in https://github.com/LibreELEC/misc-firmware/pull/9 ,i contributed this firmware and config to linux 6.6-rc , first to fw gsl repository and next to mailinglist and systemd quirks for rotation

**Comment by chewitt** [comment]
Please squash the [five] commits to a single commit and rename the single commit (not the pull-request) to `linux: add silead touchscreen driver to generic`

**Comment by chewitt** [comment]
NB: I don't see the firmware in the upstream linux-firmware repo: https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/ - can you please provide a link to the mailing list discussion?

**Comment by stpf99** [comment]
by now sorry but i cannot push to github by cmdline 
fw contributed to [gls-onitake](https://github.com/onitake/gsl-firmware/pull/215) and later  only proper code changes to linux, and systemd for rotation
https://github.com/Kaz205/linux/commit/34c271e778c1d8589ee9c833eee5ecb6fbb03149 and later to 6.5-stable was backported
https://github.com/systemd/systemd/pull/29268

**Comment by stpf99** [comment]
libreelec can overlay firmware from /storage/.config/firmware/silead directory for this fw, so only including modules should be enough and some information on howto site for users or on forum with tablet devices which uses silead

**Comment by stpf99** [comment]
i prepared test img for this modules, silead.ko i have in image but touchscreen_dmi.ko not , i have question about https://www.kernelconfig.io/config_touchscreen_dmi , should all Depends and Selects must be  listed in  linux.x86_64.conf , and depends and selects for each one from any above parents must be listed ??? and this should be in proper places ?

**Comment by stpf99** [comment]
i prepared new linux.x86_64.conf by "make menuconfig"

File: projects/Generic/linux/linux.x86_64.conf [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/x86 6.6.0-rc5 Kernel Configuration
+# Linux/x86 6.6.7 Kernel Configuration
 #
-CONFIG_CC_VERSION_TEXT="x86_64-libreelec-linux-gnu-gcc-13.2.0 (GCC) 13.2.0"
+CONFIG_CC_VERSION_TEXT="gcc (Debian 13.2.0-7) 13.2.0"
```
Comment (chewitt) [inline projects/Generic/linux/linux.x86_64.conf:5]: drop this change

File: projects/Generic/linux/linux.x86_64.conf [hunk 2]
```diff
@@ -810,7 +810,6 @@ CONFIG_ARCH_HAS_GCOV_PROFILE_ALL=y
 # end of GCOV-based kernel profiling
 
 CONFIG_HAVE_GCC_PLUGINS=y
-# CONFIG_GCC_PLUGINS is not set
```
Comment (chewitt) [inline projects/Generic/linux/linux.x86_64.conf:813]: drop this change

**Comment by chewitt** [comment]
I pushed changes to your repo with the commits squashed and renamed. I'm not sure why this closed the PR, but it may have something to do with you working in the master branch not a topic branch.

**Comment by stpf99** [comment]
LibreELEC-2:~ # dmesg |grep silead
[    6.977512] silead_ts i2c-MSSL1680:00: supply vddio not found, using dummy regulator
[    6.978283] silead_ts i2c-MSSL1680:00: supply avdd not found, using dummy regulator
[    6.978633] silead_ts i2c-MSSL1680:00: Silead chip ID: 0x80360000
[   13.025551] silead_ts i2c-MSSL1680:00: Initialization error, status: 0x0
[  252.727557] silead_ts i2c-MSSL1680:00: supply vddio not found, using dummy regulator
[  252.727856] silead_ts i2c-MSSL1680:00: supply avdd not found, using dummy regulator
[  252.729841] silead_ts i2c-MSSL1680:00: Silead chip ID: 0x80360000
[  260.257882] silead_ts i2c-MSSL1680:00: Initialization error, status: 0x0

**Comment by stpf99** [comment]
i have bow modules enablement in topic branch and PR is on https://github.com/LibreELEC/LibreELEC.tv/pull/8483

**Pull request closed without merging**